src/main/scala/io/iohk/ethereum/jsonrpc/client/RpcBaseClient.scala

```diff
@@ -0,0 +1,102 @@
+package io.iohk.ethereum.jsonrpc.client
+
+import java.io.{PrintWriter, StringWriter}
+import java.util.UUID
+
+import akka.actor.ActorSystem
+import akka.http.scaladsl.model._
+import akka.http.scaladsl.unmarshalling.Unmarshal
+import akka.http.scaladsl.{ConnectionContext, Http, HttpsConnectionContext}
+import io.circe.generic.auto._
+import io.circe.parser.parse
+import io.circe.syntax._
+import io.circe.{Decoder, Json}
+import io.iohk.ethereum.jsonrpc.JsonRpcError
+import io.iohk.ethereum.utils.Logger
+import javax.net.ssl.SSLContext
+import monix.eval.Task
+
+import scala.concurrent.duration._
+import scala.concurrent.{Await, ExecutionContext}
+
+abstract class RpcBaseClient(node: Uri, maybeSslContext: Option[SSLContext])(implicit
+    system: ActorSystem,
+    ec: ExecutionContext
+) extends Logger {
+
+  import RpcBaseClient._
+
+  lazy val connectionContext: HttpsConnectionContext =
+    maybeSslContext.fold(Http().defaultClientHttpsContext)(ConnectionContext.httpsClient)
+
+  def shutdown(): Unit = {
+    Await.ready(system.terminate(), 5.seconds)
```

Comment:
It will be better if e use timeout from config (ex. `mantis.shutdown-timeout`)

Reply:
BTW I can't find where it is used and I am not sure that this code should be able to terminate `system` passed to the class from outside.

Reply:
I am not sure if we need this function called “shutdown()”. What do you think? Do we need it?

Reply:
I think we don't need it. It should be done at some higher level class

Reply:
This function  called “shutdown” was used only in the “mallet” module. But mallet now uses the class “RpcClientMallet”. This client needs the “shutdown” function.

But we have pending removal of this module in task “ETCM-423”. So I removed this function that is not used.